provider/resource_keycloak_openid_client.go

```diff
@@ -211,6 +214,37 @@ func resourceKeycloakOpenidClient() *schema.Resource {
 				Optional: true,
 				Default:  true,
 			},
+			"extra_config": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				// you aren't allowed to specify any keys in extra_config that could be defined as top level attributes
+				ValidateDiagFunc: func(v interface{}, path cty.Path) diag.Diagnostics {
+					var diags diag.Diagnostics
+
+					extraConfig := v.(map[string]interface{})
+					value := reflect.ValueOf(&keycloak.IdentityProviderConfig{}).Elem()
```

Comment:
We need to validate the `extra_config` fields against the `keycloak.OpenidClientAttributes` type here, since that's what we use when talking to the Keycloak API for openid clients.

```suggestion
					value := reflect.ValueOf(&keycloak.OpenidClientAttributes{}).Elem()
```